crates/swc_ecma_utils/src/function/function_wrapper.rs

```diff
@@ -0,0 +1,394 @@
+        let function_ident = fn_expr.ident.take();
+        Self {
+            binding_ident: None,
+            function_ident,
+            params: Self::get_params(fn_expr.function.params.iter()),
+            function: fn_expr.into(),
+        }
+    }
+}
+
+impl From<FnDecl> for FunctionWrapper {
+    fn from(mut fn_decl: FnDecl) -> Self {
+        let function_ident = Some(fn_decl.ident.take());
+        Self {
+            binding_ident: None,
+            function_ident,
+            params: Self::get_params(fn_decl.function.params.iter()),
+            function: FnExpr {
+                ident: None,
+                function: fn_decl.function,
+            }
+            .into(),
+        }
+    }
+}
+
+impl From<ArrowExpr> for FunctionWrapper {
+    fn from(
+        ArrowExpr {
+            span,
+            params,
+            body,
+            is_async,
+            is_generator,
+            ..
+        }: ArrowExpr,
+    ) -> Self {
+        let body = Some(match body {
+            BlockStmtOrExpr::BlockStmt(block) => block,
+            BlockStmtOrExpr::Expr(expr) => BlockStmt {
+                span: DUMMY_SP,
+                stmts: vec![Stmt::Return(ReturnStmt {
+                    span: expr.span(),
+                    arg: Some(expr),
+                })],
+            },
+        });
+
+        let function = Function {
+            span,
+            params: params.into_iter().map(Into::into).collect(),
+            decorators: Default::default(),
+            body,
+            type_params: None,
+            return_type: None,
+            is_generator,
+            is_async,
+        };
+
+        let fn_expr = FnExpr {
+            ident: None,
+            function,
+        };
+
+        Self {
+            binding_ident: None,
+            function_ident: None,
+            params: Self::get_params(fn_expr.function.params.iter()),
+            function: fn_expr.into(),
+        }
+    }
+}
+
+impl TryFrom<VarDeclarator> for FunctionWrapper {
+    type Error = ();
+
+    fn try_from(var: VarDeclarator) -> Result<Self, Self::Error> {
+        if let VarDeclarator {
+            name: Pat::Ident(BindingIdent {
+                id: binding_ident, ..
+            }),
+            init: Some(init),
+            ..
+        } = var
+        {
+            match *init {
+                Expr::Fn(fn_expr) => {
+                    let mut wrapper = Self::from(fn_expr);
+                    wrapper.binding_ident = Some(binding_ident);
+
+                    Ok(wrapper)
+                }
+
+                Expr::Arrow(arrow_expr) => {
+                    let mut wrapper = Self::from(arrow_expr);
+                    wrapper.binding_ident = Some(binding_ident);
+
+                    Ok(wrapper)
+                }
+
+                _ => Err(()),
+            }
+        } else {
+            Err(())
+        }
+    }
+}
+
+impl FunctionWrapper {
+    fn get_params<'a, T, P>(params_iter: T) -> Vec<Param>
```

Comment:
Can you add a comment that this method is used to preserve `length` of a function?